src/batch-submitter/state-batch-submitter.ts

```diff
@@ -167,20 +168,33 @@ export class StateBatchSubmitter extends BatchSubmitter {
     startBlock: number,
     endBlock: number
   ): Promise<Bytes32[]> {
-    const batch: Bytes32[] = []
-    for (let i = startBlock; i < endBlock; i++) {
-      const block = (await this.l2Provider.getBlockWithTransactions(
-        i
-      )) as L2Block
+    const blockRange = endBlock - startBlock
+    const batch: Bytes32[] = await bPromise.map([...Array(blockRange).keys()], (i) => {
+      return this.l2Provider.getBlockWithTransactions(startBlock + i) as L2Block
+    }, { concurrency: 50 }).each((block: L2Block) => {
+      console.log(`block: ${block.number}`)
       if (block.transactions[0].from === this.fraudSubmissionAddress) {
-        batch.push(
-          '0xbad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1'
-        )
         this.fraudSubmissionAddress = 'no fraud'
-      } else {
-        batch.push(block.stateRoot)
+        return '0xbad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1bad1' 
       }
-    }
+
+      return block.stateRoot
+    })
```

Comment:
I'm not concertain that the `.map().each()` chain preserves the order of the original blocks since the `.each()` may happen as each resolves and this part isn't super well tested - lmk thoughts on whether this change is worth! happy to also test this

Reply:
ended up deleting this because it was breaking integration tests, so i'll try making the change separately developing against the integration repo